poetry/packages/project_package.py

```diff
@@ -0,0 +1,24 @@
+from typing import TYPE_CHECKING
+
+from poetry.core.packages.project_package import ProjectPackage as _ProjectPackage
+
+
+if TYPE_CHECKING:
+    from typing import Optional
+    from typing import Union
+
+    from poetry.core.semver.version import Version  # noqa
+
+
+class ProjectPackage(_ProjectPackage):
+    def set_version(
+        self, version, pretty_version=None
+    ):  # type: (Union[str, Version], Optional[str]) -> ProjectPackage
```

Comment:
This can move to new type hinting syntax.

Reply:
Done!